pkg/stub/stub.go

```diff
@@ -592,7 +593,12 @@ func (stub *stub) connClosed() {
 		return
 	}
 
-	os.Exit(0)
+	// the connection has closed, make a best effort to inform the plugin author about it
+	select {
+	case stub.srvErrC <- ErrNoService:
```

Comment:
hmm, this is not making the plugin to exit?

Reply:
@klihub if we loose the connection shouldn't we exit the Run method with an error?

Reply:
hmm, it returns 

>  ttrpc: server closed

and Run returns with nil

Reply:
> Maybe this was/is a really bad choice, but the reasoning behind this default behavior has been that if you don't install an OnClose handler, then you are not actively watching for the connection state so probably you won't realize that you lost connection and either should give up and exit or try to reconnect.

this contradicts the behavior of Run, so if the connection is closed , Run() exists with nil but the handlers exits the program entirely

Reply:
> > Maybe this was/is a really bad choice, but the reasoning behind this default behavior has been that if you don't install an OnClose handler, then you are not actively watching for the connection state so probably you won't realize that you lost connection and either should give up and exit or try to reconnect.
> 
> this contradicts the behavior of Run, so if the connection is closed , Run() exists with nil but the handlers exits the program entirely

@aojea Yes, you are right. It's much better to not `os.Exit()` on connection close even if there is no OnClose handler and instead return any error we might get from `Run()`.